File: app/src/main/java/com/nytimes/android/external/register/PermissionHandler.java
package com.nytimes.android.external.register;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.support.v4.app.ActivityCompat;
import android.support.v4.content.PermissionChecker;

import static android.Manifest.permission.READ_EXTERNAL_STORAGE;
import static android.support.v4.content.PermissionChecker.PERMISSION_GRANTED;

public final class PermissionHandler {

    static final int PERMISSION_REQ_CODE = 43242;

    private PermissionHandler() {
        // utility class
    }

    static void handlePermissionResult(int requestCode, Activity activity, int... grantResults) {
        if (requestCode == PERMISSION_REQ_CODE && grantResults.length > 0 && grantResults[0] == PERMISSION_GRANTED) {
            Intent intent = activity.getIntent();
            activity.finish();
            activity.startActivity(intent);
        }
    }

    public static boolean hasPermission(Context context) {
        return PermissionChecker.checkSelfPermission(context, READ_EXTERNAL_STORAGE) == PERMISSION_GRANTED;
    }

    public static void requestPermission(Activity activity) {
        ActivityCompat.requestPermissions(activity, new String[]{READ_EXTERNAL_STORAGE}, PERMISSION_REQ_CODE);
    }
}

File: app/src/main/java/com/nytimes/android/external/register/PermissionHandler.kt
package com.nytimes.android.external.register

import android.Manifest.permission.READ_EXTERNAL_STORAGE
import android.app.Activity
import android.content.Context
import androidx.core.app.ActivityCompat
import androidx.core.content.PermissionChecker
import androidx.core.content.PermissionChecker.PERMISSION_GRANTED

class PermissionHandler {

    companion object {

        const val PERMISSION_REQ_CODE = 43242

        @JvmStatic
        fun handlePermissionResult(requestCode: Int, activity: Activity, vararg grantResults: Int) {
            if (requestCode == PERMISSION_REQ_CODE &&
                    grantResults.isNotEmpty() &&
                    grantResults[0] == PERMISSION_GRANTED) {
                val intent = activity.intent
                activity.finish()
                activity.startActivity(intent)
            }
        }

        @JvmStatic
        fun hasPermission(context: Context): Boolean {
            return PermissionChecker.checkSelfPermission(context, READ_EXTERNAL_STORAGE) == PERMISSION_GRANTED
        }

        @JvmStatic
        fun requestPermission(activity: Activity) {
            ActivityCompat.requestPermissions(activity, arrayOf(READ_EXTERNAL_STORAGE), PERMISSION_REQ_CODE)
        }
    }

}
